application/Filters/DebugToolbar.php

```diff
@@ -42,16 +42,19 @@ public function after(RequestInterface $request, ResponseInterface $response)
 			$toolbar = Services::toolbar(new App());
 			$stats   = $app->getPerformanceStats();
 
-			return $response->appendBody(
-				$toolbar->run(
-					$stats['startTime'],
-					$stats['totalTime'],
-					$stats['startMemory'],
-					$request,
-					$response
-				)
-			);
-		}
+                        $output = $toolbar->run(
+                                $stats['startTime'], $stats['totalTime'], $stats['startMemory'], $request, $response
+                        );
+
+                        helper('filesystem');
+                        helper('text');
+                        $fileName = 'debugbar_' . random_string();
+                        write_file('./' . $fileName, $output, 'w+');
```

Comment:
What about to save in a [Session Flashdata](https://bcit-ci.github.io/CodeIgniter4/libraries/sessions.html#flashdata)?

Reply:
Flashdata and a custom route could have been a good alternative to the temporary files, but it would imply another call to the whole framework just to make use of the session. I mean, running around 60 php files just to serve the toolbar...

Reply:
Uhm, then a custom route and writing inside the `writable` folder.

Reply:
Yes,  the writable folder is a better place than the public folder! I did it in a hurry, just to prove the concept. Feel free to make the changes to see if we can come up with a sleek solution.

Reply:
I have tried using the Flashdata to store the toolbar but then I realized that the user can destroy the session at any time loosing the toolbar.